core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/color/SystemBarsColor.kt

```diff
@@ -1,19 +1,24 @@
 package app.k9mail.core.ui.compose.theme.color
 
+import android.app.Activity
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.SideEffect
 import androidx.compose.ui.graphics.Color
-import com.google.accompanist.systemuicontroller.rememberSystemUiController
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.platform.LocalView
+import androidx.core.view.WindowCompat
 
 @Composable
 fun SetSystemBarsColor(
-    color: Color,
+    darkTheme: Boolean,
+    color: Color = Color.Transparent,
 ) {
-    val systemUiController = rememberSystemUiController()
-
-    DisposableEffect(Unit) {
-        systemUiController.setSystemBarsColor(color = color)
-
-        onDispose { }
+    val view = LocalView.current
+    if (!view.isInEditMode) {
+        SideEffect {
```

Comment:
Why this change from `DisposableEffect`? With `SideEffect` the lambda is executed on every recomposition. 

(In practice this doesn't seem to make a difference right now.)

Reply:
My understanding is the following:

`SideEffect` is executed when its parent Composable is recomposed -> as it's part of the theme it will react to theme changes or activty related recompositions. So it's not critical.

But if you would like to reduce recompositions to a minimum `DisposableEffect` and a key of `darkTheme`and `color` would be indeed better.

I think in the theme, the impact is barely noticable. If you would like to control the status bar from a Composable deeper in the hierarchy it will recompose too often. I'll update the approach.

Reply:
Ah I remember why I picked SideEffect, it's lambda is not of composable type and requires some tweaks:

```
@Composable
fun SetSystemBarsColor(
    darkTheme: Boolean,
    color: Color = Color.Transparent,
) {
    val view = LocalView.current

    DisposableEffect(view, darkTheme, color) {
        if (!view.isInEditMode) {
            val window = (view.context as Activity).window
            window.statusBarColor = color.toArgb()
            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = !darkTheme
        }
        onDispose { }
    }
}
```

Reply:
I will stick with the original version